src/xtgeo/grid3d/_grid3d_utils.py

```diff
@@ -142,11 +153,15 @@ def _scan_ecl_keywords(
 
 
 def _scan_ecl_keywords_w_dates(
-    pfile: _XTGeoFile, maxkeys: int = MAXKEYWORDS, dataframe: bool = False
+    pfile: _XTGeoFile,
+    maxkeys: int = MAXKEYWORDS,
+    dataframe: bool = False,
 ) -> list[KeywordDateTuple] | pd.DataFrame:
     """Add a date column to the keyword"""
     xkeys = _scan_ecl_keywords(pfile, maxkeys=maxkeys, dataframe=False)
```

Comment:
using typing.overload on the `_scan_ecl_keywords(...)` and `scan_dates(...)` would remove the need for an assert here.